app/src/main/java/com/example/detodito/Login.kt

```diff
@@ -0,0 +1,138 @@
+package com.example.detodito
+
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import com.example.detodito.databinding.ActivityLoginBinding
+import com.google.firebase.FirebaseApp
+import com.google.firebase.auth.FirebaseAuth
+import com.google.gson.Gson
+import kotlinx.coroutines.*
+import okhttp3.*
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
+import okio.IOException
+import org.json.JSONObject
+
+class Login : AppCompatActivity() {
+
+    private lateinit var binding: ActivityLoginBinding
+    private val job = Job()
+    private val coroutineScope = CoroutineScope(Dispatchers.Main + job)
+    private lateinit var auth: FirebaseAuth
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        FirebaseApp.initializeApp(this)
+        auth = FirebaseAuth.getInstance()
+
+        val dialogView = LayoutInflater.from(this).inflate(R.layout.dialog_loading, null)
+
+        val dialog = AlertDialog.Builder(this)
+            .setView(dialogView)
+            .setCancelable(false)
+            .create()
+
+
+        val (myUID, state) = loadData()
+
+        if (state == true) {
+            dialog.show()
+            val intent = Intent(applicationContext, MainActivity::class.java)
+            intent.putExtra("myUID", myUID)
+            dialog.dismiss()
+            startActivity(intent)
+            finish()
+        }
+
+        binding.loginButton.setOnClickListener {
```

Comment:
binding.loginButton.setOnClickListener {
    val email = binding.emailEdittext.text.toString()
    val password = binding.passwordEdittext.text.toString()

    // Verificar si los campos de email y contraseña están vacíos
    if (email.isEmpty()) {
        binding.emailLayout.error = "Ingresa un correo válido"
        return@setOnClickListener
    }

    if (password.isEmpty()) {
        binding.passwordLayout.error = "Ingresa una contraseña"
        return@setOnClickListener
    }

    // Mostrar el diálogo de carga
    dialog.show()

    // Iniciar sesión con email y contraseña
    try {
        auth.signInWithEmailAndPassword(email, password)
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    val user = FirebaseAuth.getInstance().currentUser

                    // Obtener el token de autenticación del usuario
                    user?.getIdToken(true)?.addOnCompleteListener { task ->
                        if (task.isSuccessful) {
                            val idToken = task.result?.token

                            // Guardar el token en alguna parte
                            if (idToken != null) {
                                shared(idToken, true)
                            }

                            val intent = Intent(applicationContext, MainActivity::class.java)
                            intent.putExtra("token", idToken)

                            // Ocultar el diálogo de carga
                            dialog.dismiss()

                            // Iniciar la actividad principal
                            startActivity(intent)
                            Toast.makeText(
                                baseContext, "Autenticación exitosa.",
                                Toast.LENGTH_SHORT
                            ).show()

                            // Finalizar la actividad actual
                            finish()
                        } else {
                            // Ocultar el diálogo de carga y mostrar un mensaje de error
                            dialog.dismiss()
                            Toast.makeText(
                                baseContext, "Falló la autenticación.",
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                    }
                } else {
                    // Mostrar un mensaje de error si la autenticación falla
                    Toast.makeText(
                        baseContext, "Falló la autenticación.",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }
    } catch (e: Exception) {
        // Mostrar un mensaje de error en caso de excepción
        Toast.makeText(
            baseContext, "Falló la autenticación.",
            Toast.LENGTH_SHORT
        ).show()
        println(e.toString())
    }
}